tools/dbTools.py

```diff
@@ -1,54 +1,177 @@
+            (
+                id SERIAL PRIMARY KEY,
+                name VARCHAR(64),
+                uuid VARCHAR(64),
+                status VARCHAR(255),
+                end_time TIMESTAMP,
+                start_time TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+            );
+            """
+        )
+        self.connection.commit()
+
+    def reset_scraped(self):
+        """Set all the articles `scrap_again` attribute to 1, forcing the web
+        scraper to download and analyse them again.
+        """
+        self._execute(
+            "UPDATE article SET scrap_again = %s",
+            ('True',)
+        )
+
+    def clear_db(self):
+        """Remove all the articles from the database."""
+        self._execute(
+            "DELETE FROM article"
+        )
 
     def is_scraped(self, file_hash):
+        """Check if an article had already been scraped by looking for its file
+        hash into the database. Is the `scrap_again` attribute is true, return
+        False anyway.
+        """
         self._execute(
-            "SELECT id FROM article WHERE file_hash = ?",
+            """
+            SELECT id
+            FROM article
+            WHERE file_hash = %s AND scrap_again IS NOT TRUE;
+            """,
             (file_hash,)
         )
         result = self.cursor.fetchone()
         return result or None
 
+    def get_articles(self, offset=0, limit=-1):
+        """Return a list of articles. By default, returns every articles. This
+        method accepts start and end arguments to paginate results.
+        """
+        if limit > 0:
+            self._execute(
+                "SELECT title, file_hash, url FROM article LIMIT %s OFFSET %s",
+                (offset, limit,)
+            )
+        else:
+            self._execute("SELECT title, file_hash, url FROM article")
+        result = []
+        for article in self.cursor.fetchall():
+            result.append(article)
+        return result
+
     def insert_article(self, title, file_hash, url):
+        """Try to insert an article, composed by its title, file hash and url,
+        into the database.
+        """
+
+        if len(title) > 255:
+            self.logger.warning(
+                f'Article title ({title}) is too long ({len(title)}/255).'
+            )
+            title = title[:255]
+
+        if len(url) > 255:
+            self.logger.warning(
+                f'Article title ({url}) is too long ({len(url)}/255).'
+            )
+            url = url[:255]
         self._execute(
-            "INSERT INTO article (title, file_hash, url) VALUES (?, ?, ?)",
+            "INSERT INTO article (title, file_hash, url) VALUES (%s, %s, %s)",
             (title, file_hash, url)
         )
+
+    def get_finished_crawls(self):
+        self._execute("SELECT * FROM spiders WHERE status = %s", ('finished',))
+        result = []
+        for article in self.cursor:
+            result.append(article)
+        return result
```

Comment:
do we want to log the case where the crawl is empty?

Reply:
The more I think about it, the more I think this is important, for example in case one of our data sources changes its design. We should be noticed that the scraper became useless... Will add a task in the backlog to find a good way to deal with this case.